### PR TITLE
ICP-1974 Amazon Alexa - Wrong dimming slider behaviour when using voice commands

### PR DESCRIPTION
**Pull request by Syerramsetti915**
Changes made as per the ticket.

### CODE REVIEW

File: devicetypes/smartthings/lifx-color-bulb.src/lifx-color-bulb.groovy [hunk 1]
```diff
@@ -145,15 +145,23 @@ def setLevel(percentage) {
 	log.debug "setLevel ${percentage}"
 	if (percentage < 1 && percentage > 0) {
 		percentage = 1 // clamp to 1%
-	}
+	} else {
+    	try {
```
Comment (larsfinander) [inline devicetypes/smartthings/lifx-color-bulb.src/lifx-color-bulb.groovy]: whitespace wrong for this section (all comments applies to both files)

File: devicetypes/smartthings/lifx-color-bulb.src/lifx-color-bulb.groovy [hunk 2]
```diff
@@ -145,15 +145,23 @@ def setLevel(percentage) {
...
+        	percentage = Math.round(percentage)
+        } catch (Exception ex) {
+            log.Error "Caught exception while converting value '$percentage' to integer: $ex"
```
Comment (larsfinander) [inline devicetypes/smartthings/lifx-color-bulb.src/lifx-color-bulb.groovy]: should be "log.error", not capital "E". Not sure if catching exception is even necessary.

File: devicetypes/smartthings/lifx-color-bulb.src/lifx-color-bulb.groovy [hunk 3]
```diff
@@ -145,15 +145,23 @@ def setLevel(percentage) {
...
+        	percentage = 0
+        }
+    }
+    log.debug "setlevel: using percentage value of $percentage"
```
Comment (larsfinander) [inline devicetypes/smartthings/lifx-color-bulb.src/lifx-color-bulb.groovy]: whitespace

**Review by larsfinander — DISMISSED** [review]

File: smartapps/smartthings/lifx-connect.src/lifx-connect.groovy [hunk 4]
```diff
@@ -460,8 +460,8 @@ void updateDevices() {
 			childDevice.sendEvent(name: "saturation", value: device.color.saturation * 100)
 		}
 		childDevice.sendEvent(name: "label", value: device.label)
-		childDevice.sendEvent(name: "level", value: Math.round((device.brightness ?: 1) * 100))
-		childDevice.sendEvent(name: "switch.setLevel", value: Math.round((device.brightness ?: 1) * 100))
+		childDevice.sendEvent(name: "level", value: Math.round((device.brightness != null ? device.brightness : 1) * 100))
```
Comment (larsfinander) [inline smartapps/smartthings/lifx-connect.src/lifx-connect.groovy]: Original lines means the same as what you changed it to. It is just  a short form.

File: smartapps/smartthings/lifx-connect.src/lifx-connect.groovy [hunk 5]
```diff
@@ -460,8 +460,8 @@ void updateDevices() {
...
+		// childDevice.sendEvent(name: "switch.setLevel", value: Math.round((device.brightness ?: 1) * 100))//----------------------------------------------------------
```
Comment (larsfinander) [inline smartapps/smartthings/lifx-connect.src/lifx-connect.groovy]: Whats up with //----------------------------------------------------------

File: smartapps/smartthings/lifx-connect.src/lifx-connect.groovy (hunk 4 above)
Comment (surfous) [reply smartapps/smartthings/lifx-connect.src/lifx-connect.groovy]: Not quite true. 0 has groovy truth of false which should not be defaulted to 1 (full brightness)

File: smartapps/smartthings/lifx-connect.src/lifx-connect.groovy (hunk 5 above)
Comment (surfous) [reply smartapps/smartthings/lifx-connect.src/lifx-connect.groovy]: Line 464 needs to be deleted. I explained setLevel being a command vs. level being an attribute

**Review by surfous — DISMISSED** [review]
Remaining issues fixed, whitespace converted to spaces in all files, squashed commits. Looks good. 👍

File: devicetypes/smartthings/lifx-color-bulb.src/lifx-color-bulb.groovy [hunk 6]
```diff
@@ -5,238 +5,247 @@
...
+    
+    if (percentage == 0) {
+        return off() // if the brightness is set to 0, just turn it off
+    }
+    parent.logErrors(logObject:log) {
+        def resp = parent.apiPUT("/lights/${selector()}/state", ["brightness": percentage / 100, "power": "on"])
+        if (resp.status < 300) {
+            sendEvent(name: "level", value: percentage)
+            // sendEvent(name: "switch.setLevel", value: percentage)
```
Comment (rohandesai) [inline devicetypes/smartthings/lifx-color-bulb.src/lifx-color-bulb.groovy]: should we get rid of this if we are not using it anymore?

**Review by surfous — DISMISSED** [review]
On latest commit:
* issues addressed
* commented code removed
* whitespace normalized to spaces
* indentation fixed
* commits squashed

**Review by surfous — APPROVED** [review]
Latest commit merged in more recent changes fromupstream - conflicts were whitespace only.

**Pull request merged**